jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenterTest.java

```diff
@@ -15,290 +15,31 @@
-    private TaskCommentsView view;
+@RunWith(GwtMockitoTestRunner.class)
+public class TaskCommentsPresenterTest extends AbstractTaskCommentsPresenterTest {
 
     @Mock
-    private User identity;
+    private AbstractTaskCommentsPresenter.TaskCommentsView viewMock;
 
-    //Thing under test
     private TaskCommentsPresenter presenter;
 
     @Override
-    public AbstractTaskPresenter getPresenter() {
+    public AbstractTaskCommentsPresenter getPresenter() {
         return presenter;
     }
 
     @Before
     public void setupMocks() {
-        //Mock that actually calls the callbacks
-        callerMock = new CallerMock<>(commentsService);
-
-        presenter = new TaskCommentsPresenter(view,
-                                              callerMock,
-                                              identity);
-    }
-
-    @Test
-    public void commentsUpdatedWhenTaskSelectedOrRefreshed() {
-        String serverTemplateId = "serverTemplateId";
-        String containerId = "containerId";
-        boolean isForLog = false;
-        TaskSelectionEvent event = new TaskSelectionEvent(serverTemplateId,
-                                                          containerId,
-                                                          TASK_ID,
-                                                          "task",
-                                                          true,
-                                                          isForLog,
-                                                          "description",
-                                                          new Date(),
-                                                          "Ready",
-                                                          "actualOwner",
-                                                          2,
-                                                          1L,
-                                                          "processId");
-        //When task selected
-        presenter.onTaskSelectionEvent(event);
-
-        //Then comments for given task loaded & comment grid refreshed
-        verify(commentsService).getTaskComments(serverTemplateId,
-                                                containerId,
-                                                TASK_ID);
-        verify(view).redrawDataGrid();
-
-        //When task Refreshed
-        presenter.onTaskRefreshedEvent(new TaskRefreshedEvent(serverTemplateId,
-                                                              containerId,
-                                                              TASK_ID));
-
-        //Then comments for given task loaded & comment grid refreshed
-        verify(commentsService,
-               times(2)).getTaskComments(serverTemplateId,
-                                         containerId,
-                                         TASK_ID);
-        verify(view,
-               times(2)).redrawDataGrid();
-    }
-
-    @Test
-    public void emptyCommentNotAccepted() {
```

Comment:
do we have replacement for this?

Reply:
Now, that validation is covered at view test TaskCommentsViewImplTest with fieldValidationErrorTest and fieldValidationSuccessTest